### PR TITLE
go-mod: remove unecessary go module helper scripts

### PR DESCRIPTION
**Pull request by aanm**
Scripts under contrib/go-mod don't provide any value to Cilium
repository as the dependencies are properly managed by go modules.

This commit removed all requires that are not necessary and left only
the needed requires. After removing non-essential requires `go mod tidy &&
go mod vendor` were executed to re-create the vendor directory with the
indirect dependencies resolved.

Signed-off-by: André Martins <andre@cilium.io>

---
This change is [Reviewable](https://reviewable.io/reviews/cilium/cilium/10221)

### CODE REVIEW

**Review by joestringer — APPROVED** [review]

**Comment by aanm** [comment]
test-me-please

**Comment by aanm** [comment]
test-me-please

**Comment by aanm** [comment]
test-missed-k8s

**Comment by aanm** [comment]
test-docs-please

**Comment by aanm** [comment]
test-me-please

**Comment by aanm** [comment]
test-me-please

**Comment by jrajahalme** [comment]
@aanm These changes look confusing, any idea why they were triggered here?

`go.mod`:
```
-	github.com/miekg/dns v1.1.4
+	github.com/miekg/dns v1.0.14
...
 	github.com/miekg/dns => github.com/cilium/dns v1.1.4-0.20190417235132-8e25ec9a0ff3
```

`vendor\modules.txt`:
```
-# github.com/miekg/dns v1.1.4 => github.com/cilium/dns v1.1.4-0.20190417235132-8e25ec9a0ff3
+# github.com/miekg/dns v1.0.14 => github.com/cilium/dns v1.1.4-0.20190417235132-8e25ec9a0ff3
```

**Comment by jrajahalme** [comment]
One more note: I kind of liked the instructions on top of `go.mod`, could we have them back (updated to work without the scripts, of cause)?

**Comment by aanm** [comment]
> @aanm These changes look confusing, any idea why they were triggered here?
> 
> `go.mod`:
> 
> ```
> -	github.com/miekg/dns v1.1.4
> +	github.com/miekg/dns v1.0.14
> ...
>  	github.com/miekg/dns => github.com/cilium/dns v1.1.4-0.20190417235132-8e25ec9a0ff3
> ```
> 
> `vendor/modules.txt`:
> 
> ```
> -# github.com/miekg/dns v1.1.4 => github.com/cilium/dns v1.1.4-0.20190417235132-8e25ec9a0ff3
> +# github.com/miekg/dns v1.0.14 => github.com/cilium/dns v1.1.4-0.20190417235132-8e25ec9a0ff3
> ```

@jrajahalme the `miekg/dns` was updated to that version because that's the minimal version of `miekg/dns` the direct dependencies require. For that particular dependency it does not matter because we are overwriting it with whichever version we have in `github.com/cilium/dns`

> One more note: I kind of liked the instructions on top of `go.mod`, could we have them back (updated to work without the scripts, of cause)?

there are no more scripts. The instructions were updated in the dev documentation https://docs.cilium.io/en/latest/contributing/development/dev_setup/#add-update-a-golang-dependency

TLDR to add/update a new dependency one simply needs to do:
```
$ go get github.com/containernetworking/cni@v0.5.2
$ go mod tidy
$ go mod vendor
$ git add go.mod go.sum vendor/
```
